conda_build/variants.py

```diff
@@ -694,21 +701,22 @@ def get_package_variants(recipedir_or_metadata, config=None, variants=None):
     return filter_combined_spec_to_used_keys(combined_spec, specs=specs)
 
 
-def get_vars(variants, loop_only=False):
+@deprecated.argument("24.5", "24.7", "loop_only")
+def get_vars(variants: Iterable[dict[str, Any]]) -> set[str]:
```

Comment:
Opted to deprecate `loop_only` since `get_vars` is only invoked from one place and for that call `loop_only=True` (see `conda_build.metadata.MetaData.get_loop_vars`).